src/lib/pages/backoffice/Document/DocumentForms/DocumentEditor/DocumentForm/components/MetadataExtensions.js

```diff
@@ -25,41 +27,98 @@ const getFormComponent = fieldType => {
   }
 };
 
+const sortByLine = (a, b) => {
+  if (a.line < b.line) {
+    return -1;
+  }
+  if (a.line > b.line) {
+    return 1;
+  }
+  return 0;
+};
+
 export const MetadataExtensions = ({ extensions }) => {
   const { label, fields } = invenioConfig.DOCUMENTS.extensions;
   const configDefaults = {};
   _keys(fields).map(key => (configDefaults[key] = fields[key]['default']));
   const allExtensions = _merge(configDefaults, extensions);
 
   let components = [];
+  let orderedComponents = [];
+  let lineComponents = [];
+  let oldLine = 1;
+
+  //Create an array with the components to display
   _forOwn(allExtensions, (value, key) => {
     const componentType = invenioConfig.DOCUMENTS.extensions.fields[key].type;
-    const Component = getFormComponent(componentType);
-
-    components.push(
-      <Component
-        fieldPath={`extensions.${key}`}
-        key={key}
-        label={_get(
-          invenioConfig,
-          `extensions.document.fields.${key}.label`,
-          key
-        )}
-        optimized
-        required={_get(
-          invenioConfig,
-          `extensions.document.fields.${key}.isRequired`,
-          false
-        )}
-      />
-    );
+    const line = invenioConfig.DOCUMENTS.extensions.fields[key].line;
+    const label = invenioConfig.DOCUMENTS.extensions.fields[key].label;
+    if (componentType === 'vocabulary') {
+      components.push({
+        component: (
+          <VocabularyField
```

Comment:
I would move the creation of the components outside of this function to make it a bit more readable.